backend/pkg/modules/internal/analytics_core/init.go

```diff
@@ -0,0 +1,191 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package analytics_core
+
+import (
+	"context"
+	"net/http"
+	"sort"
+
+	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/core"
+	"github.com/openclarity/apiclarity/backend/pkg/pubsub"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	moduleVersion        = "0.0.0"
+	TraceTopicName       = "trace"
+	ApiTopicName         = "api"
+	ApiEndpointTopicName = "api_endpoint"
+	ObjectTopicName      = "object"
+	EntityTopicName      = "entity"
+)
+
+type TopicType string
+
+type AnalyticsCore struct {
+	httpHandler         http.Handler
+	msgBroker           *pubsub.Handler
+	accessor            core.BackendAccessor
+	info                *core.ModuleInfo
+	numWorkers          int
+	proccFuncRegistered map[TopicType][]AnalyticsModuleProccFunc
```

Comment:
Concurrent accesses to this map must be protected with mutexes.
From https://go.dev/blog/maps#TOC_6.
> [Maps are not safe for concurrent use](https://go.dev/doc/faq#atomic_maps): it’s not defined what happens when you read and write to them simultaneously. If you need to read from and write to a map from concurrently executing goroutines, the accesses must be mediated by some kind of synchronization mechanism. One common way to protect maps is with [sync.RWMutex](https://go.dev/pkg/sync/#RWMutex).

Reply:
Yes. This is true for map in any language..

This map shall not be updated when used. All the adding of topic and workers shall be done in initialization phase. Probably a note shall be added